test/integration/models/linode/interfaces/test_interfaces.py

```diff
@@ -0,0 +1,343 @@
+
+    def __assert_vpc(iface: LinodeInterface):
+        __assert_base(iface)
+
+        assert not iface.default_route.ipv4
+        assert not iface.default_route.ipv6
+
+        assert iface.vpc.vpc_id == vpc.id
+        assert iface.vpc.subnet_id == subnet.id
+
+        assert ipaddress.ip_address(
+            iface.vpc.ipv4.addresses[0].address
+        ) in ipaddress.ip_network(subnet.ipv4)
+        assert iface.vpc.ipv4.addresses[0].primary
+        assert iface.vpc.ipv4.addresses[0].nat_1_1_address is None
+
+        assert len(iface.vpc.ipv4.ranges) == 0
+
+    def __assert_vlan(iface: LinodeInterface):
+        __assert_base(iface)
+
+        assert not iface.default_route.ipv4
+        assert not iface.default_route.ipv6
+
+        assert iface.vlan.vlan_label == "test-vlan"
+        assert iface.vlan.ipam_address == "10.0.0.5/32"
+
+    __assert_public(instance.interfaces[0])
+    __assert_vpc(instance.interfaces[1])
+    __assert_vlan(instance.interfaces[2])
+
+    instance.invalidate()
+
+    __assert_public(instance.interfaces[0])
+    __assert_vpc(instance.interfaces[1])
+    __assert_vlan(instance.interfaces[2])
+
+
+@pytest.fixture
+def linode_interface_public(
+    test_linode_client,
+    e2e_test_firewall,
+    linode_with_interface_generation_linode,
+):
+    instance: Instance = linode_with_interface_generation_linode
+
+    ipv6_range = test_linode_client.networking.ipv6_range_allocate(
+        64, linode=instance.id
+    )
+
+    yield instance.interface_create(
+        firewall_id=e2e_test_firewall.id,
+        default_route=LinodeInterfaceDefaultRouteOptions(
+            ipv4=True,
+            ipv6=True,
+        ),
+        public=LinodeInterfacePublicOptions(
+            ipv4=LinodeInterfacePublicIPv4Options(
+                addresses=[
+                    LinodeInterfacePublicIPv4AddressOptions(
+                        address=instance.ips.ipv4.public[0].address,
+                        primary=True,
+                    )
+                ]
+            ),
+            ipv6=LinodeInterfacePublicIPv6Options(
+                ranges=[
+                    LinodeInterfacePublicIPv6RangeOptions(
+                        range=ipv6_range.range,
+                    )
+                ]
+            ),
+        ),
+    ), instance, ipv6_range
+
+
+@pytest.fixture
+def linode_interface_vpc(
+    test_linode_client,
+    e2e_test_firewall,
+    linode_with_interface_generation_linode,
+    create_vpc_with_subnet,
+):
+    instance: Instance = linode_with_interface_generation_linode
+    vpc, subnet = create_vpc_with_subnet
+
+    yield instance.interface_create(
+        firewall_id=e2e_test_firewall.id,
+        default_route=LinodeInterfaceDefaultRouteOptions(
+            ipv4=True,
+        ),
+        vpc=LinodeInterfaceVPCOptions(
+            subnet_id=subnet.id,
+            ipv4=LinodeInterfaceVPCIPv4Options(
+                # TODO (Enhanced Interfaces): Not currently working as expected
```

Comment:
I was running into something that seemed like an API bug here; will report when I have some time